app/Supports/CrawlerUrl.php

```diff
@@ -492,4 +492,30 @@ public function getUrlDepth($url)
         $splitSegment = explode('/', trim($urlPath, '/'));
         return count($splitSegment);
     }
+
+    /**
+     * Encode url except -_/? character
+     *
+     * @param  string $url []
+     * @return string
+     */
+    public function encodeUrl($url)
+    {
+        $urlTrimmed = str_replace(['http://', 'https://'], '', $url);
+        
+        $qMarkExplode = explode('?', $urlTrimmed);
+        $slashExplode = explode('/', $qMarkExplode[0]);
+        foreach ($slashExplode as $slash) {
+            $slashArray[] = urlencode($slash);
+        }
+        $qMarkExplode[0] = implode('/', $slashArray);
+        unset($slashArray);
+
+        $qMarkImplode = implode('?', $qMarkExplode);
+
+        if (strpos($url, 'http') === false) {
+            return $qMarkImplode;
+        }
+        return 'http://' . $qMarkImplode;
```

Comment:
There are two rare vulnerabilities in this function

1. If the https website don't redirect http to https then not found will be occured. I think, we must use the url's scheme instead of hardcoded with `http://`. *Currently I didn't find https websites which don't redirect **http** to **https**.*
1. If there is `'http://'` in query string which is not url encoded, then the query string will be altered. We don't want this to happen. e.g `https://www.google.co.id/?q=https://suitmedia.com`

Cause the two case are rare should I merge for now and create new issue for enhancement?